sklearn/metrics/pairwise.py

```diff
@@ -646,19 +647,33 @@ def pairwise_distances_argmin_min(
     """
     X, Y = check_pairwise_arrays(X, Y)
 
-    if metric_kwargs is None:
-        metric_kwargs = {}
-
     if axis == 0:
         X, Y = Y, X
 
-    indices, values = zip(
-        *pairwise_distances_chunked(
-            X, Y, reduce_func=_argmin_min_reduce, metric=metric, **metric_kwargs
+    if metric_kwargs is None:
+        metric_kwargs = {}
+
+    if ArgKmin.is_usable_for(X, Y, metric):
+        values, indices = ArgKmin.get_for(
+            X=X, Y=Y, k=1, metric=metric, metric_kwargs=metric_kwargs
+        ).compute(strategy="auto", return_distance=True)
+        values = np.ndarray.flatten(values)
+        indices = np.ndarray.flatten(indices)
```

Comment:
I don't understand those `np.ndarray.flatten` lines. It's weird to call a class method on an instance. What is the shape of values and indices return by `ArgKmin.compute`?

Reply:
Methods should definitely be used there.

`ArgKmin.compute` returns arrays of shape `(n_test_samples, k)`. In the case of argmin (implemented using `ArgKmin` for k=1), we need to flatten them.